net/download/service/src/main/java/de/danoeh/antennapod/net/download/service/feed/FeedUpdateManagerImpl.java

```diff
@@ -79,6 +80,7 @@ public void runOnce(Context context, Feed feed, boolean nextPage) {
         workRequest.setInputData(builder.build());
         WorkManager.getInstance(context).enqueueUniqueWork(WORK_ID_FEED_UPDATE_MANUAL,
                 ExistingWorkPolicy.REPLACE, workRequest.build());
+        SyncService.syncImmediately(context);
```

Comment:
This creates a dependency from the download service on the sync service. Please use the SynchronizationQueueSink indirection instead to keep the two modules of the app decoupled. I realize now that this class needs a new "syncImmediately" method then, so instead of a 1-line change, this would be maybe 10-20 lines then :)

Reply:
ok, will check in the next few days

Reply:
> 这会从下载服务对同步服务产生依赖关系。请使用SynchronizationQueueSink间接，以保持应用程序的两个模块分离。我现在意识到，这个类需要一个新的“立即同步”方法，所以这可能不是1行更改，而是10-20行:)



> This creates a dependency from the download service on the sync service. Please use the SynchronizationQueueSink indirection instead to keep the two modules of the app decoupled. I realize now that this class needs a new "syncImmediately" method then, so instead of a 1-line change, this would be maybe 10-20 lines then :)

Hey mate, I’ve got a rough version up and running—a new afterRefreshSync() added to SynchronizationQueueSink to manage syncing and reduce the coupling. Right now, though, there’s no retry if the initial sync fails (because it still sticking with syncNow()->SyncService(appcontent)).

It’s just a first pass, so I’m open to feedback. Do you reckon it should have a retry mechanism, or do you have another approach in mind? Let me know—I’m happy to tidy things up further. Thanks for taking the time to review this!